clang/labs/lab1/kurdina_julia/AddWarning.cpp

```diff
@@ -0,0 +1,62 @@
+#include "clang/AST/ASTConsumer.h"
+#include "clang/AST/RecursiveASTVisitor.h"
+#include "clang/Frontend/CompilerInstance.h"
+#include "clang/Frontend/FrontendPluginRegistry.h"
+
+using namespace clang;
+
+class AddWarningConsumer : public ASTConsumer {
+  CompilerInstance &Instance;
+  bool withoutClass;
+
+public:
+  AddWarningConsumer(CompilerInstance &Instance, bool withoutClass)
+      : Instance(Instance), withoutClass(withoutClass) {}
+
+  void HandleTranslationUnit(ASTContext &context) override {
+
+    struct Visitor : public RecursiveASTVisitor<Visitor> {
+      ASTContext *context;
+      bool withoutClass;
+      Visitor(ASTContext *context, bool withoutClass)
+          : context(context), withoutClass(withoutClass) {}
+
+      bool VisitFunctionDecl(FunctionDecl *FD) {
+        if (!withoutClass || withoutClass && !FD->isCXXClassMember()) {
```

Comment:
`||` already means that withoutClass == true
```suggestion
        if (!withoutClass || !FD->isCXXClassMember()) {
```